webui/src/pages/repositories/index.jsx

```diff
@@ -146,14 +151,18 @@ const RepositoriesPage = () => {
         (prefix) => router.push({pathname: `/repositories`, query: {prefix}})
     );
 
-    const createRepo = async (repo) => {
+    const createRepo = async (repo = true, presentRepo = true) => {
         try {
-            await repositories.create(repo)
+            await repositories.create(repo);
             setRefresh(!refresh);
```

Comment:
what do you mean? it refreshes the state every time a repo is created...
(also not part of the PR, but still makes sense to me)

Reply:
But the only reason to refresh is if we are going to render something different.  The only way we can render something different is if some state changed.  So...

* If only `refresh` changed, then we _will_ re-render but AFAIU we will re-render the exact same thing!
* If some other hook changed, then that hook will trigger a re-render.

As you can see, I do not understand.

Reply:
Also, a future PR should probably `useEffect` here.  Otherwise if creating a repository takes a long time, the user might click several times, and get a confusing error at some later time.  Instead, I think we should disable the button during the async call... which is basically `useEffect` :-)

Reply:
The `refresh` state is a part of the `RepositoriesPage` component. 
When creating a new repository we change the state of `refresh` which is also passed to the `RepositoryList` component which in turn uses it as a dependency for `useAPIWithPagination` changes (which uses `useEffect` internally). This method is used to fetch the newly created repo, attach it to the repository list, and present it to the user once it's finished creating.

I don't understand which button you are referring to. While creating a repository the user isn't able to click anything (look at the image in the PR's description).